GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -612,7 +599,9 @@ private Result CreatePlaceholders(string directoryRelativePath, IEnumerable<Proj
                 else
                 {
                     sha = fileInfo.Sha.ToString();
-                    fileSystemResult = this.WritePlaceholderFile(childRelativePath, fileInfo.Size, sha);
+
+                    // Writing placeholders on Mac does not require a file size
+                    fileSystemResult = this.WritePlaceholderFile(childRelativePath, endOfFile: 0, sha: sha);

```

Comment:
Add a `const` named something like `None` or `NotUsed` and maybe the value should be `-1`?